connect/runtime/src/test/java/org/apache/kafka/connect/runtime/rest/RestClientTest.java

```diff
@@ -0,0 +1,128 @@
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.apache.kafka.connect.runtime.WorkerConfig;
+import org.apache.kafka.connect.runtime.rest.entities.ErrorMessage;
+import org.apache.kafka.connect.runtime.rest.errors.ConnectRestException;
+import org.eclipse.jetty.client.HttpClient;
+import org.eclipse.jetty.client.api.ContentResponse;
+import org.eclipse.jetty.client.api.Request;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import javax.ws.rs.core.Response;
+import java.util.Collections;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeoutException;
+
+import static org.easymock.EasyMock.anyString;
+import static org.easymock.EasyMock.expect;
+import static org.easymock.EasyMock.niceMock;
+import static org.easymock.EasyMock.replay;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNull;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+public class RestClientTest {
+
+    private static final ObjectMapper OBJECT_MAPPER = new ObjectMapper();
+    private static final TypeReference<TestDTO> TEST_TYPE = new TypeReference<TestDTO>() {
+    };
+    private HttpClient httpClient;
+
+    private static String toJsonString(Object obj) {
+        try {
+            return OBJECT_MAPPER.writeValueAsString(obj);
+        } catch (JsonProcessingException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    private static <T> RestClient.HttpResponse<T> httpRequest(HttpClient httpClient, TypeReference<T> typeReference) {
+        return RestClient.httpRequest(
+                httpClient, null, null, null, null, typeReference, null, null);
+    }
+
+    @BeforeEach
+    public void mockSetup() {
+        httpClient = niceMock(HttpClient.class);
+    }
+
+    @Test
+    public void testSuccess() throws ExecutionException, InterruptedException, TimeoutException {
+        int statusCode = Response.Status.OK.getStatusCode();
+        String expectedResponse = toJsonString(new TestDTO("someContent"));
+        setupHttpClient(statusCode, expectedResponse);
+
+        RestClient.HttpResponse<TestDTO> httpResp = httpRequest(httpClient, TEST_TYPE);
+        assertEquals(httpResp.status(), statusCode);
+        assertEquals(toJsonString(httpResp.body()), expectedResponse);
+    }
+
+    @Test
+    public void testNoContent() throws ExecutionException, InterruptedException, TimeoutException {
+        int statusCode = Response.Status.NO_CONTENT.getStatusCode();
+        setupHttpClient(statusCode, null);
+
+        RestClient.HttpResponse<TestDTO> httpResp = httpRequest(httpClient, TEST_TYPE);
+        assertEquals(httpResp.status(), statusCode);
+        assertNull(httpResp.body());
+    }
+
+    @Test
+    public void testError() throws ExecutionException, InterruptedException, TimeoutException {
```

Comment:
This test case is about more than just handling errors in the client, right? We might want to rename it to something like `testErrorResponseStatusGetsPreserved` and then possibly add other cases that cover `IOException`s, unexpected (i.e, 1XX or 3XX) HTTP status codes, failures during request signing, and other possible `RuntimeException`s that may be encountered in the body of `RestClient::httpRequest`.

Reply:
It'd also be fine to leave these other cases out for now since you've strictly increased the coverage for this class with the cases you've added already, but I do think it'd still be worth adding specificity to the name of the test case here just to illustrate to future maintainers what the test covers and what it does not.

Reply:
@C0urante I'm curios about the reason for throwing a ConnectRestException with 500 error code when the client receives an unexpected 1xx or 3xx, only logging the actual 1xx/3xx code. Do you have a clue why ConnectRestException doesn't throw the actual error code? Is this an issue and should it be addressed in this PR?

Reply:
I think a 500 in that case is intentional, since we don't know how to handle those statuses gracefully and there's nothing in the code base that would cause them to be returned. I haven't seen any issues from this logic, either personally or on the Jira for the project. It's probably fine to leave as-is.

Reply:
I've added tests for the cases you've mentioned.
sending the request may trigger 3 different exceptions, so i've put them in a parameterised test, but in junit 4 you can't have a single parameterised test case - the whole class runs once for each parameter.
I've used [this](https://stackoverflow.com/a/35057629) approach to isolate that test case, but IMO the test class became a little hard to read.

Do you know any better way to isolate that one parameterised test case? I know JUnit5 can do that but it's not in the classpath unfortunately

there could also just be 3 separate test cases, but a parameterised test seems like the perfect fit here

Reply:
I think we should hold off on using Junit 5 for now; these changes are fine as-are.